### ISSUE TITLE
False Positive for Unused Label

### ISSUE THREAD

**Issue by JackStouffer** [Hackerpilot/Dscanner#363]
From std/internal/math/biguintx86.d line 732

```
uint multibyteMulAdd(char op)(uint [] dest, const uint [] src, uint
        multiplier, uint carry) pure {
    // Timing: This is the most time-critical bignum function.
    // Pentium M: 5.4 cycles/operation, still has 2 resource stalls + 1load block/iteration

    // The main loop is pipelined and unrolled by 2,
    //   so entry to the loop is also complicated.

    // Register usage
    // EDX:EAX = multiply
    // EBX = counter
    // ECX = carry1
    // EBP = carry2
    // EDI = dest
    // ESI = src

    enum string OP = (op=='+')? "add" : "sub";
    version(D_PIC)
    {
        enum { zero = 0 }
    }
    else
    {
        // use p2 (load unit) instead of the overworked p0 or p1 (ALU units)
        // when initializing registers to zero.
        __gshared int zero = 0;
        // use p3/p4 units
        __gshared int storagenop; // write-only
    }

    enum { LASTPARAM = 5*4 } // 4* pushes + return address.
    asm pure nothrow {
        naked;

        push ESI;
        push EDI;
        push EBX;
        push EBP;
        mov EDI, [ESP + LASTPARAM + 4*4]; // dest.ptr
        mov EBX, [ESP + LASTPARAM + 4*3]; // dest.length
        align 16;
        nop;
        mov ESI, [ESP + LASTPARAM + 4*2];  // src.ptr
        lea EDI, [EDI + 4*EBX]; // EDI = end of dest
        lea ESI, [ESI + 4*EBX]; // ESI = end of src
        mov EBP, 0;
        mov ECX, EAX; // ECX = input carry.
        neg EBX;                // count UP to zero.
        mov EAX, [ESI+4*EBX];
        test EBX, 1;
        jnz L_enter_odd;
    }
    // Main loop, with entry point for even length
    mixin("asm pure nothrow {" ~ asmMulAdd_innerloop(OP, "ESP+LASTPARAM") ~ "}");
    asm pure nothrow {
        mov EAX, EBP; // get final carry
        pop EBP;
        pop EBX;
        pop EDI;
        pop ESI;
        ret 5*4;
    }
L_enter_odd:
    mixin("asm pure nothrow {" ~ asmMulAdd_enter_odd(OP, "ESP+LASTPARAM") ~ "}");
}
```

Gives

```
std/internal/math/biguintx86.d(794:1)[warn]: Label "L_enter_odd" is not used.
```

**Comment by Hackerpilot** [comment]
Of course it's inline ASM...

**Comment by BBasile** [comment]
It was fixed by #393 (#392 was a dup). You can close.
